test/test.js

```diff
@@ -130,6 +144,19 @@ describe('Auth', () => {
           .then(() => expect(a.authedId()).toEqual(hexStr));
       });
 
+      it('should have an error if local auth is unsuccessful', (done) => {
+        expect.assertions(3);
+        const a = new Auth(null, '/auth');
+        return a.provider('userpass').login('fake-user', 'password')
+          .then(() => done('expected an error on unsuccessful login'))
+          .catch(e => {
+            expect(e).toBeInstanceOf(Error);
+            expect(e.response.status).toBe(401);
+            expect(e.error).toBe('unauthorized');
```

Comment:
might as well assert something for `errorCode` here too, since it's being set in the response.